kafka-rest/src/main/java/io/confluent/kafkarest/resources/v3/TopicsResource.java

```diff
@@ -132,6 +134,37 @@ public void getTopic(
     AsyncResponses.asyncResume(asyncResponse, response);
   }
 
+  @PATCH
```

Comment:
Re: PATCH vs PUT
I really like the PATCH approach. My take is that unlike with updating topic configs, where we use a PUT and access a resource-like endpoint related to the modified topic config, here the API is designed to access the topic resource while providing a modification for part of the topic - its partition count. PUT would be plain incorrect with the model proposed here, but in addition to that PATCH seems to be a really good fit. My only worry is that PATCH is not as popular, understood, and maybe supported, but it has been out there for a long time now (RFC 5789 is 10+ years old) so this shouldn't be a problem.